maitake-sync/src/wait_cell.rs

```diff
@@ -327,6 +327,46 @@ impl WaitCell {
         }
     }
 
+    /// Asynchronously poll the [`WaitCell`] until a condition occurs
+    ///
+    /// This can be used to implement a "wait loop", turning a "try" function
+    /// (e.g. "try_recv" or "try_send") into an asynchronous function (e.g.
+    /// "recv" or "send").
+    ///
```

Comment:
if you have the time, it would be nice to have an example here, but no biggie --- i'm fine with merging this without one.